File: samples/JavaScript/monte-carlo-methods.js
function mcpi(n){
  var x,y,m=0;

  for(var i = 0; i < n; i += 1) {
    x = Math.random();
    y = Math.random();

    if (x*x + y*y < 1) { m += 1; }
  }

  return 4*m/n;
}

console.log(mcpi(1000));
console.log(mcpi(10000));
console.log(mcpi(100000));
console.log(mcpi(1000000));
console.log(mcpi(10000000));

File: samples/Lua/monte-carlo-methods.lua
function MonteCarlo ( n_throws )
    math.randomseed( os.time() )

    n_inside = 0
    for i = 1, n_throws do
    	if math.random()^2 + math.random()^2 <= 1.0 then
            n_inside = n_inside + 1
    	end
    end

    return 4 * n_inside / n_throws
end

print( MonteCarlo( 10000 ) )
print( MonteCarlo( 100000 ) )
print( MonteCarlo( 1000000 ) )
print( MonteCarlo( 10000000 ) )
